langchain4j-weaviate/src/main/java/dev/langchain4j/store/embedding/weaviate/WeaviateEmbeddingStore.java

```diff
@@ -146,6 +149,49 @@ public List<String> addAll(List<Embedding> embeddings, List<TextSegment> embedde
         return addAll(null, embeddings, embedded);
     }
 
+    @Override
+    public void remove(String id) {
+        ensureNotBlank(id, "id");
+        client.data().deleter().
+                withClassName(objectClass).
+                withID(id).
+                run();
+    }
+
+    @Override
+    public void removeAll(Collection<String> ids) {
+        ensureNotEmpty(ids, "ids");
+        client.batch().objectsBatchDeleter()
+                .withClassName(objectClass)
+                .withWhere(WhereFilter.builder()
+                        .path("id")
+                        .operator(Operator.ContainsAny)
+                        .valueText(ids.toArray(new String[0]))
+                        .build())
+                .run();
+    }
+
+    @Override
+    public void removeAll(Filter filter) {
+        ensureNotNull(filter, "filter");
+        WhereFilter whereFilter = WeaviateMetadataFilterMapper.map(filter);
+        client.batch().objectsBatchDeleter()
+                .withClassName(objectClass)
+                .withWhere(WhereFilter.builder()
+                        .path("metadata.id")
+                        .operator(Operator.ContainsAny)
+                        .valueText("1")
```

Comment:
Is it intended that it tries to remove all entries with metadata id 1?

Reply:
Sorry I hardcoded it for testing, I have updated the code.